src/types/bytes.rs

```diff
@@ -26,6 +26,60 @@ impl PyBytes {
         unsafe { py.from_owned_ptr(ffi::PyBytes_FromStringAndSize(ptr, len)) }
     }
 
+    /// Creates a new Python bytestring object and a mutable slice to its underlying buffer.
+    /// The bytestring is uninitialised and must not be read.
+    ///
+    /// Panics if out of memory.
+    unsafe fn new_with_uninit_impl(py: Python<'_>, len: usize) -> (&mut [u8], &PyBytes) {
+        let length = len as ffi::Py_ssize_t;
+        let pyptr = ffi::PyBytes_FromStringAndSize(std::ptr::null(), length);
+        // Iff pyptr is null, py.from_owned_ptr(pyptr) will panic
+        let pybytes = py.from_owned_ptr(pyptr);
+        let buffer = ffi::PyBytes_AsString(pyptr) as *mut u8;
+        debug_assert!(!buffer.is_null());
+        let slice = std::slice::from_raw_parts_mut(buffer, len);
+        (slice, pybytes)
+    }
+
+    /// Creates a new Python bytestring object.
+    /// The `init` closure can initialise the bytestring.
+    ///
+    /// # Safety
```

Comment:
We don't need `#Safety` document for safe functions

Reply:
Fixed in 533c105b73d214afdb14033f45b1659cf0723233